File: app/models/gig.py
from beanie import Document, Indexed, Link
from pydantic import Field
from typing import List, Optional
from datetime import datetime, timedelta
from .user import User

class Gig(Document):
    title: Indexed(str)
    description: str
    category: str = Field(..., pattern="^(Tech|Creative|Academic|Other)$")
    price: float
    price_type: str = "fixed" # fixed / negotiable
    
    # Status Management
    status: str = "open" # open | in_progress | completed
    posted_by: Link[User]
    assigned_to: Optional[Link[User]] = None
    
    # Matching & AI Tags
    skill_tags: List[str] = []
    is_alumni_eligible: bool = False
    
    # Time Tracking
    deadline: datetime
    created_at: datetime = Field(default_factory=datetime.utcnow)
    completed_at: Optional[datetime] = None

    class Settings:
        name = "gigs"
        # Indexing for faster searching on the feed
        indexes = ["status", "category", "created_at"]

File: src/models/Gig.js
import mongoose from 'mongoose'

const GigSchema = new mongoose.Schema({
  postedBy: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  title: { type: String, required: true },
  description: { type: String, default: '' },
  skillTags: { type: [String], default: [] },
  category: { type: String, enum: ['Tech', 'Creative', 'Academic'], required: true },
  price: { type: Number, required: true },
  priceType: { type: String, enum: ['fixed', 'negotiable'], default: 'fixed' },
  deadline: { type: Date },
  visibility: { type: String, enum: ['all', 'department', 'year'], default: 'all' },
  status: { type: String, enum: ['open', 'in_progress', 'completed'], default: 'open' },
  assignedTo: { type: mongoose.Schema.Types.ObjectId, ref: 'User', default: null },
  isAlumniEligible: { type: Boolean, default: false },
  applicants: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }],
}, { timestamps: true })

export default mongoose.models.Gig || mongoose.model('Gig', GigSchema)
